src/StateService/StatePlugin.cs

```diff
@@ -239,5 +242,72 @@ public JObject GetStateHeight(JArray _params)
             json["validatedrootindex"] = StateStore.Singleton.ValidatedRootIndex;
             return json;
         }
+
+        private ContractState GetHistoricalContractState(MPTTrie<StorageKey, StorageItem> trie, UInt160 script_hash)
+        {
+            const byte prefix = 8;
+            StorageKey skey = new KeyBuilder(NativeContract.ContractManagement.Id, prefix).Add(script_hash);
+            return trie[skey]?.GetInteroperable<ContractState>();
+        }
+
+        private void PrepareStateParams(JArray _params, out MPTTrie<StorageKey, StorageItem> trie, out StorageKey skey)
```

Comment:
I prefer to avoid JArray, and receive the arguments. It will be easy to know what should I send to `FindState`

Reply:
Updated.